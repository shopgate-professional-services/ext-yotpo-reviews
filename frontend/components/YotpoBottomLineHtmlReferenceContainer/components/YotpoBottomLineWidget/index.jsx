import React, { Component } from 'react';
import PropTypes from 'prop-types';
import getYotpo from '../../../../helpers/getYotpo';

/**
 * Yotpo bottom line widget
 */
class YotpoBottomLineWidget extends Component {
  static propTypes = {
    config: PropTypes.shape().isRequired,
    productId: PropTypes.string.isRequired,
    product: PropTypes.shape(),
  };

  static defaultProps = {
    product: null,
  }

  /**
   * Constructs
   * @param {Objects} props props
   */
  constructor(props) {
    super(props);
    this.yotpoMountTimeout = undefined;
  }

  /**
   * Mounts Yotpo script.
   */
  componentDidMount() {
    if (!window.yotpo) {
      getYotpo(this.props.config.yotpoAppKey);
      return;
    }
    // Clean and refreshWidgets are needs to keep yotpo object information relevant
    window.yotpo.clean();
    window.yotpo.refreshWidgets();
    this.yotpoMountTimeout = setTimeout(() => {
      window.yotpo.initWidgets();
      this.yotpoMountTimeout = undefined;
    }, 500);
  }

  /**
   * Cleanup timeout request
   */
  componentWillUnmount() {
    // Set initialized to false on unmounting
    window.yotpo.initialized = false;
    clearTimeout(this.yotpoMountTimeout);
  }

  /**
   * Renders.
   * @return {JSX}
   */
  render() {
    if (!this.props.product) {
      return null;
    }
    return (
      <div
        className="yotpo bottomLine"
        data-appkey={`${this.props.config.yotpoAppKey}`}
        data-domain=""
        data-product-id={`${this.props.productId}`}
        data-name={`${this.props.product.name}`}
        data-url={`${this.props.product.productUrl}`}
        data-image-url={`${this.props.product.featuredImageUrl}`}
        data-description=""
      />
    );
  }
}

export default YotpoBottomLineWidget;
