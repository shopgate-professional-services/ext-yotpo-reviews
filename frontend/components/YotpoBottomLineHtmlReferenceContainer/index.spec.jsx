import React from 'react';
import { mount } from 'enzyme';
import { createMockStore } from '@shopgate/pwa-common/store';
import { Provider } from 'react-redux';
import YotpoBottomLineWidget from './components/YotpoBottomLineWidget';
import HtmlReferenceContainer from './index';

jest.mock('./components/YotpoBottomLineWidget', () => () => (<div>YotpoBottomLineWidget</div>));

const store = createMockStore();

describe('HtmlReferenceContainer', () => {
  it('should render HtmlReferenceContainer', () => {
    const component = mount((
      <Provider store={store}>
        <HtmlReferenceContainer
          config={{ yotpoAppKey: 'mock' }}
          productId="product_one"
        />
      </Provider>
    ));
    expect(component).toMatchSnapshot();
    expect(component.find(YotpoBottomLineWidget).length).toBe(1);
  });
});
