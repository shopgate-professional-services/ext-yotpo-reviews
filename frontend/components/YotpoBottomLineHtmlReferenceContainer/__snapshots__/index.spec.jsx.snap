// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HtmlReferenceContainer should render HtmlReferenceContainer 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <HtmlReferenceContainer
    config={
      Object {
        "yotpoAppKey": "mock",
      }
    }
    product={Object {}}
    productId="product_one"
  >
    <div>
      <Component
        config={
          Object {
            "yotpoAppKey": "mock",
          }
        }
        product={Object {}}
        productId="product_one"
      >
        <div>
          YotpoBottomLineWidget
        </div>
      </Component>
    </div>
  </HtmlReferenceContainer>
</Provider>
`;
